src/init/features/dataconnect/sdk.ts

```diff
@@ -150,17 +150,12 @@ async function askQuestions(setup: Setup, config: Config): Promise<SDKInfo> {
     // app/src/main/kotlin and app/src/main/java are conventional for Android,
     // but not required or enforced. If one of them is present (preferring the
     // "kotlin" directory), use it. Otherwise, fall back to the app directory.
-    let baseDir = path.join(appDir, "generated");
-    for (const candidateSubdir of ["app/src/main/java", "app/src/main/kotlin"]) {
-      const candidateDir = path.join(appDir, candidateSubdir);
-      if (fs.existsSync(candidateDir)) {
-        baseDir = candidateDir;
-      }
-    }
```

Comment:
This isn't needed. @dconeybe and I made it always append the package path after the output path.

Just did the same for Swift as well.

Reply:
Did you chat about this recently? @dconeybe requested this behavior in comments on https://github.com/firebase/firebase-tools/pull/7522

Reply:
Yeah, this was fairly recent.

He actually added this to emulator.

Reply:
@fredzqm, this logic was added by me in https://github.com/firebase/firebase-tools/pull/7535. If there is no `app/src/main/java` or `app/src/main/kotlin` then it creates a folder named "generated" and puts the connector into its own directory under there. If, however, one of those two directories _do_ exist, then they are used. This is the desirable behavior. The other SDKs also go into a "generated" directory, so it made sense, to me, to put them _all_ in there. Hopefully customers will change it anyways.

Reply:
There must be some confusion here. The preference of using `app/src/main/...` is a special case that follows the conventional folder structure of basic Android apps. If you're going to remove this logic, please remove the comment above it as well.

Reply:
I think the was some miscommunication here. AFAICT, we want to keep this logic - #7547 adds it back in.

Separately, even after that tweak, the output looks redundant/weird to me - by default, the connectorID shows up 2x in the output path
<img width="445" alt="Screenshot 2024-08-09 at 6 45 57 AM" src="https://github.com/user-attachments/assets/3e0453f9-cdb2-4200-9352-b3f3520f6f6a">

Reply:
Discussed offline. Reverted that change.

Thanks for catching it!